docs/source/getting_started.rst

```diff
@@ -300,34 +300,144 @@ syntax like so:
+Since ``AuditedBankAccount`` is a subclass of ``BankAccount``, we can directly pass
+in instances of it into our ``transfer`` function:
 
 .. code-block:: python
 
-  x = chr(4)
+    audited = AuditedBankAccount('Charlie', 300)
+    transfer(account_1, audited, 100)   # Type checks!
+
+This behavior is actually a fundamental aspect of the PEP 484 type system: when 
+we annotate some variable with a type ``T``, we are actually telling mypy that
+variable can be assigned an instance of ``T``, or an instance of a *subclass* of ``T``.
+The same rule applies to type hints on parameters or fields.
+
+See :ref:`class-basics` to learn more about how to work with code involving classes.
+
+
+.. _stubs-intro:
+
+Stubs files and typeshed
+************************
+
+Mypy also understands how to work with classes found in the standard library.
+For example, here is a function which uses the ``Path`` object from the
+`pathlib standard library module <https://docs.python.org/3/library/pathlib.html>`_:
+
+.. code-block:: python
+
+    from pathlib import Path
+
+    def load_template(template_path: Path, name: str) -> str:
+        # Mypy understands that 'file_path.read_text()' returns a str...
+        template = template_path.read_text()
+
+        # ...so understands this line type checks.
+        return template.replace('USERNAME', name)
+
+This behavior may surprise you if you're familiar with how
+Python internally works. The standard library does not use type hints
+anywhere, so how did mypy know that ``Path.read_text()`` returns a ``str``,
+or that ``str.replace(...)`` accepts exactly two ``str`` arguments?
+
+The answer is that mypy comes bundled with *stub files* from the
+the `typeshed <https://github.com/python/typeshed>`_ project.
+
+A *stub file* is a file containing a skeleton of the public interface
+of that Python module, including classes, variables, functions -- and
+most importantly, their types. Typeshed is a collection of stub files
+for modules in the standard library and select third party libraries
+and is what mypy used to identify the correct types for our example above.
+
+Mypy can understand third party libraries in the same way as long as those
+libraries come bundled with stub files. (See :ref:`installed-packages`
+for more details on how this is done). However, if the third party library
+does *not* come bundled with type hints, mypy will not try and guess what
+the types are: it'll assume the entire library is
+:ref:`dynamically typed <dynamic-typing>` and report an error
+whenever you import the library.
+
+We discuss strategies for handling this category of errors in
+:ref:`ignore-missing-imports`, but in short, there are three things you can do:
+
+1. Look for a separate package containing stubs for that library. For example,
+   if you want type hints for the `sqlalchemy <https://www.sqlalchemy.org/>`_ 
+   library, you can install
+   `sqlalchemy-stubs <https://pypi.org/project/sqlalchemy-stubs/>`_.
```

Comment:
Should we also mention other notable stubs projects? This link appears twice (here and in the section on running mypy), maybe replace one of them with e.g. `django-stubs`?